Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -555,3 +586,91 @@ void ABenchmarkGymGameModeBase::ReportAuthoritativePlayers_Implementation(const
 		}
 	}
 }
+
+void ABenchmarkGymGameModeBase::TickActorMigration(float DeltaSeconds)
+{
+	if (bHasActorMigrationCheckFailed)
+	{
+		return;
+	}
+
+	const UNFRConstants* Constants = UNFRConstants::Get(GetWorld());
+	check(Constants);
+
+	// This test respects the initial delay timer in both native and GDK
+	if (Constants->ActorMigrationCheckDelay.HasTimerGoneOff())
+	{
+		// Count how many actors hand over authority in 1 tick
+		int MovementAuthActors = GetMovementAuthActors();
+		int Delta = FMath::Abs(MovementAuthActors - PreviousTickMigration);
+		ToBeRemovedMigrationDeltas.Push(Delta);
+		int RemoveValue = 0;
+		bool bChanged = false;
+		if (MigrationFrameCount > MigrationWindowFrameCount)
+		{
+			RemoveValue = ToBeRemovedMigrationDeltas[0];
+			ToBeRemovedMigrationDeltas.RemoveAt(0);
+			MigrationOfCurrentWorker -= RemoveValue;
+			bChanged = (Delta != RemoveValue);
+		}
+		MigrationOfCurrentWorker += Delta;
+		PreviousTickMigration = MovementAuthActors;
+		++MigrationFrameCount;
+
+		if (MigrationFrameCount > MigrationWindowFrameCount)
+		{
+			if (bChanged)
+			{
+				// Only report MigratedActorsOfCurrentWorker to the worker which has authority and the Migration changed
+				ReportMigration(FPlatformProcess::ComputerName(), MigrationOfCurrentWorker);
+			}
+
+			if (HasAuthority() && ActorMigrationCheckTimer.HasTimerGoneOff())
+			{
+				if (MapWorkerActorMigration.Num() > 1)
+				{
+					double TotalMigrations = 0;
+					for (const auto& KeyValue : MapWorkerActorMigration)
+					{
+						TotalMigrations += KeyValue.Value;
+					}
+					double AverageActorMigration = TotalMigrations / static_cast<double>(MigrationWindowFrameCount);
+					if (AverageActorMigration < MinActorMigration)
+					{
+						bHasActorMigrationCheckFailed = true;
+						NFR_LOG(LogBenchmarkGymGameModeBase, Error, TEXT("%s: Actor migration check failed. TotalMigrations=%.3f AverageActorMigration=%.3f MinActorMigration=%.3f"),
+							*NFRFailureString, TotalMigrations, AverageActorMigration, MinActorMigration);
+					}
+					else
+					{
+						// reset timer for next check after 10s
+						ActorMigrationCheckTimer.SetTimer(10);
+					}
+				}
+			}
+		}
+	}
+}
+
+int ABenchmarkGymGameModeBase::GetMovementAuthActors() const
```

Comment:
The name of this function is misleading, What does this function return? A count of actor which have a UUserExperienceComponent? .. Perhaps GetAuthUxActorsCount() or GetCountAuthUxActors() ...? Is this function new or do we already have a function that gets the count of authoritative actors with UxComponent..?